app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt

```diff
@@ -26,19 +72,149 @@ class SplashActivityTest {
     SplashActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false
   )
 
-  @Before
-  fun setUp() {
-    Intents.init()
+  @Test
+  fun testSplashActivity_initialOpen_routesToOnboardingActivity() {
+    activityTestRule.launchActivity(null)
+    intended(hasComponent(OnboardingActivity::class.java.name))
   }
 
   @Test
-  fun testSplashActivity_initialOpen_routesToHomeActivity() {
-    activityTestRule.launchActivity(null)
-    intended(hasComponent(ProfileActivity::class.java.name))
+  fun testSplashActivity_secondOpen_routesToChooseProfileActivity() {
+    simulateAppAlreadyOpened()
+    launch(SplashActivity::class.java).use {
+      intended(hasComponent(ProfileActivity::class.java.name))
+    }
   }
 
-  @After
-  fun tearDown() {
-    Intents.release()
+  @Test
+  fun testSplashActivity_secondOpen_profileChooserScreenContentVerified() {
+    simulateAppAlreadyOpened()
+    launch(SplashActivity::class.java).use {
+      // Wait until the expected text appears on the screen, and ensure it's for the welcome text view.
+      onView(withId(R.id.profile_select_text)).check(matches(withText(R.string.profile_chooser_select)))
+    }
+  }
+
+  private fun simulateNewAppInstance() {
+    // Simulate a fresh app install by clearing any potential on-disk caches using an isolated app history controller.
+    createTestRootComponent().getOnboardingFlowController().clearOnboardingFlow()
+    onIdle()
+  }
+
+  private fun simulateAppAlreadyOpened() {
+    // Simulate the app was already opened by creating an isolated app history controller and saving the opened status
+    // on the system before the activity is opened.
```

Comment:
Update this comment

Reply:
Updated to   
  // Simulate the app was already onboarded by creating an isolated on-boarding flow controller and saving the onboarding status
    // on the system before the activity is opened.